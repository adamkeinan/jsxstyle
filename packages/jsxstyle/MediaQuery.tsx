import * as React from 'react';
import CSSProperties from './CSSProperties';
import { cache } from './jsxstyle';

export type MQProps = {
  className?: string;
  style?: React.CSSProperties;
  query: string;
} & CSSProperties;

export type OldStyleProps = {
  mediaQueries: { mq: string };
} & CSSProperties;

function wrapPropsWithMQPrefix(props: MQProps): OldStyleProps {
  const ret = { mediaQueries: { mq: props.query } };
  for (const k in props) {
    if (k === 'query' || k === 'children') continue;
    ret[`mq${k[0].toUpperCase()}${k.slice(1)}`] = props[k];
  }
  return ret;
}

export default class MediaQuery extends React.PureComponent<MQProps> {
  className: string | null;

  constructor(props: MQProps) {
    super(props);
    this.className = cache.getClassName(
      wrapPropsWithMQPrefix(props),
      this.props.className,
      true
    );
  }

  componentWillReceiveProps(props: MQProps) {
    this.className = cache.getClassName(
      wrapPropsWithMQPrefix(props),
      this.props.className,
      true
    );
  }

  render() {
    const onlyChild = React.Children.only(this.props.children);
    const props = { ...onlyChild.props, className: this.className };
    return React.cloneElement(onlyChild, props);
  }
}
