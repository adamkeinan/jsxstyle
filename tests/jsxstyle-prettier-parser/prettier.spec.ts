import jsxstyleParser = require('jsxstyle-prettier-parser');
import * as prettier from 'prettier';

const demoSource = `
import { Block } from 'jsxstyle';

<Block
  color="rgba(0,0,0,.1) "
  backgroundImage="linear-gradient( to bottom,  0% red, 100% blue )"
/>;

<div
  color="rgba(0,0,0,.1) "
  backgroundImage="linear-gradient( to bottom,  0% red, 100% blue )"
/>;
`;

describe('prettier-plugin-jsxstyle', () => {
  it('parses JS files', () => {
    const output = prettier.format(demoSource, {
      filepath: 'test.js',
      parser: jsxstyleParser,
    });

    expect(output).toMatchSnapshot();
  });

  it('parses TSX files', () => {
    const output = prettier.format(demoSource, {
      filepath: 'test.tsx',
      parser: jsxstyleParser,
    });

    expect(output).toMatchSnapshot();
  });

  it('does not parse non-TS/JS files with jsxstyle-prettier-parser', () => {
    const getOutput = () => {
      prettier.format('whatever', {
        filepath: 'test.css',
        parser: jsxstyleParser,
      });
    };

    expect(getOutput).toThrowErrorMatchingSnapshot();
  });
});
