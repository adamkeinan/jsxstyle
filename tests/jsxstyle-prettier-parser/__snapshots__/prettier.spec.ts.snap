// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`prettier-plugin-jsxstyle does not parse non-TS/JS files with jsxstyle-prettier-parser 1`] = `undefined`;

exports[`prettier-plugin-jsxstyle parses JS files 1`] = `
"import { Block } from \\"jsxstyle\\";

<Block
  color=\\"rgba(0, 0, 0, 0.1)\\"
  backgroundImage=\\"linear-gradient(to bottom, 0% red, 100% blue)\\"
/>;

<div
  color=\\"rgba(0,0,0,.1) \\"
  backgroundImage=\\"linear-gradient( to bottom,  0% red, 100% blue )\\"
/>;
"
`;

exports[`prettier-plugin-jsxstyle parses TSX files 1`] = `
"import { Block } from \\"jsxstyle\\";

<Block
  color=\\"rgba(0, 0, 0, 0.1)\\"
  backgroundImage=\\"linear-gradient(to bottom, 0% red, 100% blue)\\"
/>;

<div
  color=\\"rgba(0,0,0,.1) \\"
  backgroundImage=\\"linear-gradient( to bottom,  0% red, 100% blue )\\"
/>;
"
`;
